src/core/QCheck.ml

```diff
@@ -650,14 +650,17 @@ module Shrink = struct
 
   let unit = nil
 
-  (* balanced shrinker for integers (non-exhaustive) *)
+  (* inspired by QCheck2's int shrinker algorithm (non-exhaustive) *)
   let int x yield =
-    let y = ref x in
+    let curr = ref 0 in (*to return 0 repeatedly *)  (*was: let curr = ref (x/2) *)
     (* try some divisors *)
-    while !y < -2 || !y >2 do y := !y / 2; yield (x - !y); done; (* fast path *)
-    if x>0 then yield (x-1);
-    if x<0 then yield (x+1);
-    ()
+    while !curr <> x do
+      yield !curr;
```

Comment:
Unrelated to your merge request but what is `yield`? As `x` is continuously shrinking I guess it it's in charge of both marking the numbers of steps but also stop when the shrinker found the minimal value?